### ISSUE TITLE
hyphen in tag extras file tag list breaks build

### ISSUE THREAD

**Issue by galthaus** [squidfunk/mkdocs-material#6973]
### Bug description

When adding this to mkdocs.yml, the build breaks.
```
plugins:
  - tags:
      tags_extra_files:
        tags/object.md:
          - test-automation
 ```

Produces stack trace:

```
ERROR   -  Error reading page 'tags/object.md': bad character range t-a at position 36
Traceback (most recent call last):
  File "/home/galthaus/docs/greg/venv/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
  File "/home/galthaus/docs/greg/venv/lib/python3.8/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/home/galthaus/docs/greg/venv/lib/python3.8/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/home/galthaus/docs/greg/venv/lib/python3.8/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/galthaus/docs/greg/venv/lib/python3.8/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/galthaus/docs/greg/venv/lib/python3.8/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/home/galthaus/docs/greg/venv/lib/python3.8/site-packages/mkdocs/__main__.py", line 286, in build_command
    build.build(cfg, dirty=not clean)
  File "/home/galthaus/docs/greg/venv/lib/python3.8/site-packages/mkdocs/commands/build.py", line 322, in build
    _populate_page(file.page, config, files, dirty)
  File "/home/galthaus/docs/greg/venv/lib/python3.8/site-packages/mkdocs/commands/build.py", line 171, in _populate_page
    page.markdown = config.plugins.on_page_markdown(
  File "/home/galthaus/docs/greg/venv/lib/python3.8/site-packages/mkdocs/plugins.py", line 575, in on_page_markdown
    return self.run_event('page_markdown', markdown, page=page, config=config, files=files)
  File "/home/galthaus/docs/greg/venv/lib/python3.8/site-packages/mkdocs/plugins.py", line 507, in run_event
    result = method(item, **kwargs)
  File "/home/galthaus/docs/greg/venv/lib/python3.8/site-packages/material/plugins/tags/plugin.py", line 163, in on_page_markdown
    markdown = self._handle_deprecated_tags_extra_files(page, markdown)
  File "/home/galthaus/docs/greg/venv/lib/python3.8/site-packages/material/plugins/tags/plugin.py", line 292, in _handle_deprecated_tags_extra_files
    if not re.search(pattern, markdown):
  File "/usr/lib/python3.8/re.py", line 201, in search
    return _compile(pattern, flags).search(string)
  File "/usr/lib/python3.8/re.py", line 304, in _compile
    p = sre_compile.compile(pattern, flags)
  File "/usr/lib/python3.8/sre_compile.py", line 764, in compile
    p = sre_parse.parse(p, flags)
  File "/usr/lib/python3.8/sre_parse.py", line 948, in parse
    p = _parse_sub(source, state, flags & SRE_FLAG_VERBOSE, 0)
  File "/usr/lib/python3.8/sre_parse.py", line 443, in _parse_sub
    itemsappend(_parse(source, state, verbose, nested + 1,
  File "/usr/lib/python3.8/sre_parse.py", line 598, in _parse
    raise source.error(msg, len(this) + 1 + len(that))
re.error: bad character range t-a at position 36
```

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
-

### Reproduction

[9.5.15+insiders.4.53.3-hyphen-tags-break-build.zip](https://github.com/squidfunk/mkdocs-material/files/14746529/9.5.15%2Binsiders.4.53.3-hyphen-tags-break-build.zip)

### Steps to reproduce

1. Update mkdocs.yml
2. mkdocs build

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) using the [built-in info plugin](https://squidfunk.github.io/mkdocs-material/plugins/info/).

**Comment by alexvoss** [comment]
Thanks for reporting and for providing the reproduction. I will have a look to see if I can spot there the issue lies. 

Just so you are aware, the Insider Edition version of the tags plugin has seen a significant revision recently that allows you to produce [additional tag index pages](https://squidfunk.github.io/mkdocs-material/plugins/tags/#listing-configuration) without using `tags_extra_files`, which is now [deprecated](https://squidfunk.github.io/mkdocs-material/insiders/changelog/#4.48.0). Existing documentation should still build, of course, so will label this as a bug.

**Comment by alexvoss** [comment]
@squidfunk in [tags/plugin.py:L291](https://github.com/squidfunk/mkdocs-material-insiders/blob/473c087f6f6ff85f0974847ed3b77915a1dbb0da/src/plugins/tags/plugin.py#L291), `directive` is "`material/tags { include: [test-automation] }`", so `re` interprets the part in square brackets as a character set and `t-a` as a range. Characters with special meanings in regular expressions should be escaped before that line.

**Comment by alexvoss** [comment]
Looks like in [tags/pluging.py:L286](https://github.com/squidfunk/mkdocs-material-insiders/blob/473c087f6f6ff85f0974847ed3b77915a1dbb0da/src/plugins/tags/plugin.py#L286) what should be updated is the variable `markdown` and not `page.markdown` - cf. `_handle_deprecated_tags_file()` above.

**Comment by squidfunk** [comment]
Fixed in [`4ff6a57`](https://github.com/squidfunk/mkdocs-material-insiders/commit/4ff6a572151f684fe2e21527582e1096a88805d0) by @alexvoss 🎉

**Comment by galthaus** [comment]
Awesome.  I'll look into moving off the tag_extra_files.

**Comment by squidfunk** [comment]
Released as part of Released as part of [9.5.16+insiders-4.53.4](https://squidfunk.github.io/mkdocs-material/insiders/changelog/#4.53.4).
